warehouse/integrations/snowflake/table_manager.go

```diff
@@ -0,0 +1,128 @@
+package snowflake
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	whutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+type tableManager interface {
+	createTableQuery(schemaIdentifier, tableName string, columns model.TableSchema) string
+	addColumnsQuery(schemaIdentifier, tableName string, columnsInfo []whutils.ColumnInfo) (string, error)
+}
+
+func newTableManager(config *config.Config, warehouse model.Warehouse) tableManager {
+	if warehouse.GetBoolDestinationConfig(model.EnableIcebergSetting) {
+		externalVolume := warehouse.GetStringDestinationConfig(config, model.ExternalVolumeSetting)
+		return newIcebergTableManager(externalVolume)
+	}
+	return newStandardTableManager()
+}
+
+// for standard Snowflake tables
+type standardTableManager struct {
+	dataTypesMap map[string]string
+}
+
+func newStandardTableManager() tableManager {
+	return &standardTableManager{
+		dataTypesMap: map[string]string{
+			"boolean":  "boolean",
+			"int":      "number",
+			"bigint":   "number",
+			"float":    "double precision",
+			"string":   "varchar",
+			"datetime": "timestamp_tz",
+			"json":     "variant",
+		},
+	}
+}
+
+func (m *standardTableManager) createTableQuery(schemaIdentifier, tableName string, columns model.TableSchema) string {
+	return fmt.Sprintf(
+		`CREATE TABLE IF NOT EXISTS %s.%q ( %v )`,
+		schemaIdentifier, tableName, columnsWithDataTypes(columns, "", m.dataTypesMap),
+	)
+}
+
+func (m *standardTableManager) addColumnsQuery(schemaIdentifier, tableName string, columnsInfo []whutils.ColumnInfo) (string, error) {
+	var queryBuilder strings.Builder
+	queryBuilder.WriteString(fmt.Sprintf(`
+		ALTER TABLE
+		  %s.%q
+		ADD COLUMN`,
+		schemaIdentifier,
+		tableName,
+	))
+
+	for _, columnInfo := range columnsInfo {
+		dataType := m.dataTypesMap[columnInfo.Type]
+		queryBuilder.WriteString(fmt.Sprintf(` %q %s,`, columnInfo.Name, dataType))
```

Comment:
```suggestion
		queryBuilder.WriteString(fmt.Sprintf(`  IF NOT EXISTS %q %s,`, columnInfo.Name, dataType))
```

shouldn't this be considering as per #5581 ?

Reply:
Yes, I implemented this change before I started working on 5881. Will update the PR